SOURCES/etc/xapi.d/plugins/netdata.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# this is a XCP-ng plugin. It goes to /etc/xapi.d/plugins/ with the exec bit set.
+# it expects to have the XCP-ng python xapi plugins library installed. (https://github.com/xcp-ng/xcp-ng-xapi-plugins)
+
+import json
+import os
+import shutil
+import sys
+import tempfile
+import urllib2
+
+import XenAPIPlugin
+
+from urlparse import urlparse
+
+sys.path.append('.')
+from xcpngutils import configure_logging, run_command, error_wrapped
+from xcpngutils.filelocker import FileLocker
+
+
+class OperationLocker(FileLocker):
+    def _lock(self):
+        try:
+            # noinspection PyProtectedMember
+            super(OperationLocker, self)._lock()
+        except Exception:
+            raise Exception('The plugin is busy.')
+
+
+netdata_config_content = '''
+# netdata configuration
+# do not edit, managed by XCP-ng
+
+[global]
+    memory mode = none
+    run as user = netdata
+    history = 7200
+    bind to = localhost
+[web]
+    mode = none
+    web files owner = root
+    web files group = root
+    default port = 19999
+    bind to = localhost
+    allow netdata.conf from = localhost
+'''
+
+netdata_streaming_content = '''
+# do not edit, managed by XCP-ng
+
+[stream]
+    # Enable this on slaves, to have them send metrics.
+    enabled = yes
+    destination = {0}
+    api key = {1}
+    timeout seconds = 60
+    default port = 19999
+    send charts matching = *
+    buffer size bytes = 1048576
+    reconnect delay seconds = 5
+    initial clock resync iterations = 60
+'''
+
+# xe host-call-plugin host-uuid=a34931d6-fcc5-4f4a-9896-f03ea1ede176 plugin=netdata.py fn=install_netdata args:api_key='c1814542-c066-11e9-a752-080027d5b5e4' args:destination='tcp:192.168.0.34:19999'
+# should return JSON true or error
+@error_wrapped
+def install_netdata(session, args):
+    api_key = args['api_key']
+    destination = args['destination']
+    with OperationLocker():
+        temp_dir = tempfile.mkdtemp()
+        files = []
+        try:
+            command = ['yum', 'install', '-y', 'netdata', '--enablerepo=xcp-ng-testing']
```

Comment:
The command to be merged would be without the `--enablerepo` parameter. It's OK if the netdata package is not available yet in the repositories. We don't want to issue an update to xcp-ng-xapi-plugins to change that parameter afterwards.

By the way, what about creating a generic installation method that other plugins could reuse, like we devised earlier?

Reply:
actually, it's because of the `--enablerepo` that I think it doesn't yet make sense to extract a method.

Reply:
A method could take an optional `enablerepo` parameter, but since we don't want `--enablerepo` in the yum install call here, doesn't extracting a method make sense?